### ISSUE TITLE
Performing a `RenderPass` inside a debug group results in Vulkan validation errors

### ISSUE THREAD

**Issue by SludgePhD** [gfx-rs/wgpu#3733]
**Description**
When a `RenderPass` is performed inside of a `push_debug_group`/`pop_debug_group` pair, the following Vulkan validation error is emitted:

```
[2023-04-27T20:32:37Z ERROR wgpu_hal::vulkan::instance] VALIDATION [VUID-vkCmdEndDebugUtilsLabelEXT-commandBuffer-01912 (0x56146426)]
        Validation Error: [ VUID-vkCmdEndDebugUtilsLabelEXT-commandBuffer-01912 ] Object 0: handle = 0x563ee6e45520, type = VK_OBJECT_TYPE_COMMAND_BUFFER; | MessageID = 0x56146426 | vkCmdEndDebugUtilsLabelEXT() called without a corresponding vkCmdBeginDebugUtilsLabelEXT first The Vulkan spec states: There must be an outstanding vkCmdBeginDebugUtilsLabelEXT command prior to the vkCmdEndDebugUtilsLabelEXT on the queue that commandBuffer is submitted to (https://www.khronos.org/registry/vulkan/specs/1.3-extensions/html/vkspec.html#VUID-vkCmdEndDebugUtilsLabelEXT-commandBuffer-01912)
[2023-04-27T20:32:37Z ERROR wgpu_hal::vulkan::instance]         objects: (type: COMMAND_BUFFER, hndl: 0x563ee6e45520, name: ?)
```

**Repro steps**
Modify the `hello-triangle` example to add `encoder.push_debug_group("bla")` before and `encoder.pop_debug_group()` after the render pass.

**Expected vs observed behavior**
There should not be any validation errors.

**Extra materials**
I'm using the `vulkan-validation-layers` Arch Linux package at version 1.3.243.0-1 (recent versions add more validation, so this might not reproduce with a lower version, although I haven't tried).

**Platform**
Arch Linux, Vulkan, Wayland/X11 both repro.

**Comment by trojanfoe** [comment]
Reproduced with some code I am playing with.  It seems OK when recording a compute pass however.

Linux Mint 21.1
Vulkan SDK 1.3.243.0
wgpu 0.16.0

**Comment by Sebbl0508** [comment]
This issue also appears when using a simple bloom example with bevy `0.10.1`  
It appears in the `bloom_upsampling_pass` Renderpass.  
  
Fedora 38  
Vulkan SDK 1.3.243
wgpu 0.15.1 (used by bevy)

**Comment by jimblandy** [comment]
I think simply running `cargo nextest run --workspace --exclude cts_runner` reproduces this, too. ^^;;;

**Comment by teoxoy** [comment]
This is most likely a false positive (see https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/5671) since we create a new command buffer on `push_debug_group`

https://github.com/gfx-rs/wgpu/blob/7f4c39c8db75a7a9608394c05ad3941ef768588d/wgpu-core/src/command/mod.rs#L391 

and another when the render pass gets dropped

https://github.com/gfx-rs/wgpu/blob/7f4c39c8db75a7a9608394c05ad3941ef768588d/wgpu-core/src/command/render.rs#L1241

causing the `push/pop_debug_group`s to be on different command buffers.

**Comment by jimblandy** [comment]
In addition to the sprious VUID complaint, I also get a crash here on Mesa AMD Vulkan. The crash seems to be fixed in mesa commit [1c64952e](https://gitlab.freedesktop.org/mesa/mesa/-/commit/1c64952e65f0dc6fccabc2e036cd16e3901a31ed) (2023-3-18). That is not on the Mesa 23.0 branch, though.
